src/java/com/reucon/openfire/plugin/archive/xep0313/IQQueryHandler.java

```diff
@@ -278,26 +286,95 @@ private IQ buildForbiddenResponse(IQ packet) {
             }
         }
 
-        Date startDate = null;
-        Date endDate = null;
-        try {
-            if(startField != null) {
-                startDate = xmppDateTimeFormat.parseString(startField);
-            }
-            if(endField != null) {
-                endDate = xmppDateTimeFormat.parseString(endField);
-            }
-        } catch (ParseException e) {
-            Log.error("Error parsing query date filters.", e);
+       try
+        {
+	        LocalDate nowDate = LocalDate.now();
```

Comment:
If you use ZonedDateTime here you can avoid some of the messiness around atZoneId() at around line 307